### DISCUSSION TITLE
DAP config not working

### DISCUSSION THREAD

**Discussion by emersonfbarros** [mrcjkb/rustaceanvim#293] [Q&A]
I put the configuration code on a file that will be loaded before the rustceanvim plugin, as README instructs to do, but when a I start dap it shows the message that isn't found a dap config for rust. I already tried to put that in the config and init field on rustceanvim table in lazy config, but also not work. what can I do to get that codelldb with nvim-dap working correctly?

```
vim.g.rustaceanvim = function()
  local install_path = require('mason-registry').get_package('codelldb'):get_install_path()
  local lldb_path = install_path .. 'extension/adapter/codelldb'
  local liblldb_path = install_path .. 'extension/lldb/lib/liblldb.so'

  local cfg = require 'rustaceanvim.config'
  return {
    dap = {
      adapter = cfg.get_codelldb_adapter(lldb_path, liblldb_path),
    },
  }
end
```

**Comment by mrcjkb** [comment]
Hey :wave: 

you shouldn't need to configure anything here. If you have mason.nvim configured to make sure codelldb is installed, rustaceanvim will pick it up automatically.

For troubleshooting, see the [section in the readme](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#troubleshooting).

**Reply by emersonfbarros** [reply]
according to checkhealth everything is ok. but still get the dap message about not find rust setup :/

**Reply by mrcjkb** [reply]
Does it work with `:RustLsp debuggables`?
I can't really help much if I don't know what your project looks like.
For rust-analyzer to be able to find debug configurations, you need to open a file with test modules.
